travelpal/assets/js/components/ProfileView.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import ProfileLabel from './ProfileLabel';
+import { Button } from 'reactstrap';
+
+import api from '../api';
+
+// Renders the user's profile
+export default function ProfileView({ users, friends, username, userId }) {
+    const currentUser = _.find(users, (u) => {
+        return u.username === username
+    });
+
+    const friendRecord = _.find(friends, (f) => {
+        return (
+            (f.acceptor.id == currentUser.id && f.requestor.id == userId)
+            || (f.acceptor.id == userId && f.requestor.id == currentUser.id)
+        );
+    });
+    // :status, :requestor_id, :acceptor_id
+    function sendFriendRequest() {
+        api.friend_request({ status: "Pending", requestor_id: userId, acceptor_id: currentUser.id })
+    }
+    let btnTxt = "Add Friend";
+    let func = sendFriendRequest;
+    if (friendRecord) {
```

Comment:
Should add a functionality for "Deny Request" as well. Clicking this should delete the record.